File: app/src/main/java/me/eagzzycsl/darkroom/ui/AdapterNaughty.java
package me.eagzzycsl.darkroom.ui;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import me.eagzzycsl.darkroom.model.NaughtyApp;
import me.eagzzycsl.darkroom.R;

class AdapterNaughty extends BaseAdapter<NaughtyApp> {
    AdapterNaughty(Context context) {
        super(context);
    }

    @Override
    public int getLayoutId() {
        return R.layout.layout_item_naughty;
    }

    @Override
    public NaughtyHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new NaughtyHolder(LayoutInflater.from(context).inflate(getLayoutId(), parent, false));
    }

    private class NaughtyHolder extends RecViewHolder<NaughtyApp> {
        private final TextView item_name;
        private final ImageView item_icon;
        private final View item_add_shortcut;

        NaughtyHolder(View itemView) {
            super(itemView);
            item_name = itemView.findViewById(R.id.item_name);
            item_icon = itemView.findViewById(R.id.item_icon);
            item_add_shortcut = itemView.findViewById(R.id.item_add_shortcut);
            itemView.setOnClickListener(view -> {
                NaughtyApp naughtyApp = (NaughtyApp) itemView.getTag();
                naughtyApp.launch(context);
            });
            itemView.setOnLongClickListener(view -> {
                NaughtyApp naughtyApp = (NaughtyApp) itemView.getTag();
                naughtyApp.goToSettings(context);
                return true;
            });
            item_add_shortcut.setOnClickListener(view -> {
                NaughtyApp naughtyApp = (NaughtyApp) itemView.getTag();
                naughtyApp.createShortcut(context);
            });
        }

        @Override
        public void setContent(NaughtyApp naughtyApp) {
            itemView.setTag(naughtyApp);
            item_name.setText(naughtyApp.getAppName());
            item_icon.setImageDrawable(naughtyApp.getAppIcon());
            if (naughtyApp.getMetaApp().isEnable(context)) {
                item_name.setTextColor(context.getColor(android.R.color.black));
            } else {
                item_name.setTextColor(context.getColor(R.color.colorAccent));
            }
        }
    }
}

File: app/src/main/java/me/eagzzycsl/darkroom/ui/AdapterNaughty.kt
package me.eagzzycsl.darkroom.ui

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import me.eagzzycsl.darkroom.R
import me.eagzzycsl.darkroom.manager.AppManager
import me.eagzzycsl.darkroom.manager.ShortCutManager
import me.eagzzycsl.darkroom.model.MetaApp

internal class AdapterNaughty(context: Context) : BaseAdapter<MetaApp>(context) {

    override val layoutId: Int
        get() = R.layout.layout_item_naughty

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NaughtyHolder {
        return NaughtyHolder(LayoutInflater.from(context).inflate(layoutId, parent, false))
    }

    internal inner class NaughtyHolder internal constructor(itemView: View) : BaseAdapter<MetaApp>.RecViewHolder<MetaApp>(itemView) {
        private val itemName: TextView = itemView.findViewById(R.id.item_name)
        private val itemIcon: ImageView = itemView.findViewById(R.id.item_icon)
        private val itemAddShortcut: View = itemView.findViewById(R.id.item_add_shortcut)
        private val itemRelease: View = itemView.findViewById(R.id.item_release)

        init {
            itemView.setOnClickListener {
                val naughtyApp = itemView.tag as MetaApp
                naughtyApp.launch(context)
            }
            itemView.setOnLongClickListener {
                val naughtyApp = itemView.tag as MetaApp
                AppManager.gotoSettings(context, naughtyApp)
                true
            }
            itemAddShortcut.setOnClickListener {
                val naughtyApp = itemView.tag as MetaApp
                ShortCutManager.createShortcut(context, naughtyApp)
            }
            itemRelease.setOnClickListener {
                val naughtyApp = itemView.tag as MetaApp

            }
        }

        override fun setContent(data: MetaApp) {
            itemView.tag = data
            itemName.text = data.appName
            itemIcon.setImageDrawable(data.appIcon)
            if (data.isEnable(context)) {
                itemName.setTextColor(context.getColor(android.R.color.black))
            } else {
                itemName.setTextColor(context.getColor(R.color.colorAccent))
            }
        }
    }
}
